src/main/java/com/google/jenkins/plugins/computeengine/client/ClientUtil.java

```diff
@@ -0,0 +1,95 @@
+package com.google.jenkins.plugins.computeengine.client;
+
+import com.cloudbees.plugins.credentials.CredentialsMatchers;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.domains.DomainRequirement;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.http.HttpTransport;
+import com.google.cloud.graphite.platforms.plugin.client.ClientFactory;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.jenkins.plugins.computeengine.ComputeEngineScopeRequirement;
+import com.google.jenkins.plugins.credentials.oauth.GoogleOAuth2Credentials;
+import com.google.jenkins.plugins.credentials.oauth.GoogleRobotCredentials;
+import hudson.AbortException;
+import hudson.model.ItemGroup;
+import hudson.security.ACL;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.List;
+import java.util.Optional;
+
+/** Utilities for using the gcp-plugin-core clients. */
+public class ClientUtil {
+
+  private static final String APPLICATION_NAME = "jenkins-google-compute-plugin";
+
+  /**
+   * Creates a {@link ClientFactory} for generating the GCP api clients.
+   *
+   * @param itemGroup The Jenkins context to use for retrieving the credentials.
+   * @param domainRequirements A list of domain requirements. Must be non-null.
```

Comment:
Remove "must be non-null". All params not using the Optional<T> type are assumed to be non-null.

Reply:
Done